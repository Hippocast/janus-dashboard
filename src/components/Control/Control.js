import React from 'react';
import PropTypes from 'prop-types';

import block from '../../helpers/bem-cn';
import Icon from '../Icon/Icon';

import './Control.css';

const b = block('j-control');

const propTypes = {
    icon: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
    className: PropTypes.string,
};

const Control = ({ className, icon, onClick }) => (
    <button className={b.mix(className)} type="button" onClick={onClick}>
        <Icon className={b('icon')()} type={icon} />
    </button>
);

Control.propTypes = propTypes;

export default Control;
