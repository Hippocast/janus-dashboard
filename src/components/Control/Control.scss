@import 'index.scss';

$control-size: ceil($hf-gutter*2.4);
$control-icon-size: ceil($hf-gutter*1.6);

.j-control {
    width: $control-size;
    height: $control-size;
    border: none;
    padding: 0;
    margin: 0;
    background: none;
    cursor: pointer;

    &__icon {
        display: inline-block;
        width: 16px;
        vertical-align: middle;
    }
}
