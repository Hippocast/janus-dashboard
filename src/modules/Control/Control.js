import React from 'react';

import block from '../../helpers/bem-cn';

import './Control.css';

const b = block('j-control');

const Control = ({ type }) => {
  return (
    <span className={b({ type })}></span>
  );
};

export default Control;
