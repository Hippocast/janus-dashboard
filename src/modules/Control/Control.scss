$control-size: 20px;

$urls: (
  edit: url(../../assets/img/edit.svg),
  checked: url(../../assets/img/checked.svg),
  delete: url(../../assets/img/delete.svg),
  copy: url(../../assets/img/copy.svg),
);

.j-control {
  display: inline-block;
  width: $control-size;
  height: $control-size;
  background-repeat: no-repeat;
}

@each $type, $url in $urls {
  .j-control--type-#{$type} {
    background-image: $url;
  }
}
