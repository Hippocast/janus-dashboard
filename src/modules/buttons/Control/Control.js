import React from 'react';
import PropTypes from 'prop-types';

import block from '../../../helpers/bem-cn';
import Icon from '../../Icon/Icon';

import './Control.css';

const b = block('j-control');

const propTypes = {
    icon: Icon.propTypes.type,
    onClick: PropTypes.func.isRequired,
};

const Control = ({ className, icon, onClick }) => (
    <button className={b()} type="button" onClick={onClick}>
        <Icon className={b('icon')()} type={icon} />
    </button>
);

Control.propTypes = propTypes;

export default Control;
