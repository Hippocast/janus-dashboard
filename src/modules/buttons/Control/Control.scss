@import 'index.scss';

$control-size: ceil($hf-gutter*2.4);
$control-icon-size: ceil($hf-gutter*1.6);

.j-control {
    width: $control-size;
    height: $control-size;
    border: none;
    padding: 0;
    margin: 0;
    background: none;

    &__icon {
        display: inline-block;
        width: 16px;
        height: 16px;
        vertical-align: top;
    }
}
