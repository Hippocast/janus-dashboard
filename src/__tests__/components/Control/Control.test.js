/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'

import Control from '../../../components/Control/Control'

describe('Control component', () => {
  const render = props => renderer.create(<Control {...props} />).toJSON()
  const mockFunction = jest.fn()
  const requiredProps = {
    icon: 'add',
    onClick: mockFunction
  }
  const runCommonTest = tree => {
    const icon = tree.children[0]
    expect(tree.props.className).toContain('j-control')
    expect(icon.props.className).toContain('j-icon')
    expect(icon.props.className).toContain('j-icon--type-add')
    tree.props.onClick()
    expect(mockFunction).toHaveBeenCalled()
    expect(tree).toMatchSnapshot()
  }

  beforeEach(() => {
    mockFunction.mockReset()
  })

  it('renders correctly', () => {
    const props = requiredProps
    const tree = render(props)

    runCommonTest(tree)
  })

  it('renders with an additional className if passed', () => {
    const props = {
      ...requiredProps,
      className: 'mock-class'
    }
    const tree = render(props)

    runCommonTest(tree)
    expect(tree.props.className).toContain('mock-class')
  })

  it('renders with an additional ariaLabel if passed', () => {
    const props = {
      ...requiredProps,
      ariaLabel: 'mock-label'
    }
    const tree = render(props)
    const icon = tree.children[0]

    runCommonTest(tree)
    expect(icon.props['aria-label']).toContain(props.ariaLabel)
  })

  it('renders with an additional className and ariaLabel if passed', () => {
    const props = {
      ...requiredProps,
      className: 'mock-class',
      ariaLabel: 'mock-label'
    }
    const tree = render(props)
    const icon = tree.children[0]

    runCommonTest(tree)
    expect(tree.props.className).toContain('mock-class')
    expect(icon.props['aria-label']).toContain(props.ariaLabel)
  })
})
