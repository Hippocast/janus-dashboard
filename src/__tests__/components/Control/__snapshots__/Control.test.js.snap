// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component renders correctly 1`] = `
<button
  className="j-control"
  onClick={[Function]}
  type="button"
>
  <span
    aria-label={undefined}
    className="j-icon j-icon--type-add j-control__icon"
  />
</button>
`;

exports[`Icon component renders with an additional ariaLabel if passed 1`] = `
<button
  className="j-control"
  onClick={[Function]}
  type="button"
>
  <span
    aria-label="mock-label"
    className="j-icon j-icon--type-add j-control__icon j-tooltiped"
  />
</button>
`;

exports[`Icon component renders with an additional className and ariaLabel if passed 1`] = `
<button
  className="j-control mock-class"
  onClick={[Function]}
  type="button"
>
  <span
    aria-label="mock-label"
    className="j-icon j-icon--type-add j-control__icon j-tooltiped"
  />
</button>
`;

exports[`Icon component renders with an additional className if passed 1`] = `
<button
  className="j-control mock-class"
  onClick={[Function]}
  type="button"
>
  <span
    aria-label={undefined}
    className="j-icon j-icon--type-add j-control__icon"
  />
</button>
`;
